app/controllers/jobs_controller.rb

```diff
@@ -16,29 +25,55 @@ def create
       flash[:success] = "You created #{@job.title} at #{@company.name}"
       redirect_to company_job_path(@company, @job)
     else
+      @errors = @job.errors.full_messages
+      @categories = Category.all
+      @page_setup = PageSetup.new(:jobs_new)
       render :new
     end
   end
 
   def show
     @job = Job.find(params[:id])
+    @company = Company.find(params[:company_id])
+    @job_category = @job.category
```

Comment:
same here - we can just access this in the view. in general, you should try to set a limited amount of instance variables in your controller actions 👍